### PR TITLE
docs(drawer): implement and document attrs and vars

### PR DESCRIPTION
**Pull request by lalithkarikelli**
## Description

`hx-drawer` : Implement and document attributes and variables

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF 2104

## Before you request a review for this PR:

- [x] For UI changes, did you manually test in recent versions of modern browsers (Chrome, Firefox, and Safari)?
- [x] For UI changes, did you manually test in IE11 and legacy Edge?
- [ ] Did you add component tests for any new code?
- [ ] Did you run the component unit tests via `yarn test` to ensure all tests passed?
- [ ] Did you include a screenshot of the component tests?
- [ ] If you changed/added functionality, did you update the demo page and documentation?
- [ ] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] In Jira, have you linked to this PR on the ticket(s)?

### CODE REVIEW

**Review by BY00565233 — APPROVED** [review]
DEV LGTM

File: src/scss/components/drawer/_index.scss [hunk 1]
```diff
@@ -39,7 +39,7 @@ hx-drawer {
   }
 
   > hx-div {
-    background-color: $gray-0;
+    background-color: var(--hxDrawer-hxDiv-bgcolor, $gray-0);
     border-top: 1px solid $gray-400;
```
Comment (100stacks) [inline src/scss/components/drawer/_index.scss:43]: Please change to:

```css
border-top: 1px solid var(--hxDrawer-hxdiv-border-top, $gray-400);
```

File: src/scss/components/drawer/_index.scss [hunk 2]
```diff
@@ -48,7 +48,7 @@ hx-drawer {
   }
 
   > footer {
-    background-color: $gray-0;
+    background-color: var(--hxDrawer-footer-bgcolor, $gray-0);
     border-top: 1px solid $gray-400;
```
Comment (100stacks) [inline src/scss/components/drawer/_index.scss:52]: Please change to:

```css
border-top: 1px solid var(--hxDrawer-footer-border-top, $gray-400);
```

File: src/scss/components/drawer/_index.scss [hunk 3]
```diff
@@ -39,7 +39,7 @@ hx-drawer {
   }
 
   > hx-div {
-    background-color: $gray-0;
+    background-color: var(--hxDrawer-hxDiv-bgcolor, $gray-0);
```
Comment (100stacks) [inline src/scss/components/drawer/_index.scss:42]: Please change to:

```css
background-color: var(--hxDrawer-hxdiv-bgcolor, $gray-0);
```

**Review by 100stacks — CHANGES_REQUESTED** [review]

File: src/scss/components/drawer/_index.scss (hunk 1 above)
Comment (lalithkarikelli) [reply src/scss/components/drawer/_index.scss]: Changes have been done, please review.

File: src/scss/components/drawer/_index.scss (hunk 2 above)
Comment (lalithkarikelli) [reply src/scss/components/drawer/_index.scss]: Changes have been done, please review.

File: src/scss/components/drawer/_index.scss (hunk 3 above)
Comment (lalithkarikelli) [reply src/scss/components/drawer/_index.scss]: Changes have been done, please review.

**Review by 100stacks — APPROVED** [review]
Dev LGTM

**Pull request merged**